From 69df551dfed7f7bdcd59e462e7c7558d1f180cef Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Sun, 7 Mar 2021 17:15:20 +0200
Subject: [PATCH 03/19] (halium) never set FLAT_BINDER_FLAG_TXN_SECURITY_CTX
 flag

It breaks binder transactions when run on a kernel without
SELinux support.

Change-Id: Ibff0f58a7867061a9353bf8bf5ab092ed562b65f
---
 libs/binder/Parcel.cpp       | 5 +++--
 libs/binder/ProcessState.cpp | 3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index c1770b35d1..c2c30c02eb 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -306,9 +306,10 @@ status_t Parcel::flattenBinder(const sp<IBinder>& binder) {
                 schedBits = schedPolicyMask(policy, priority);
             }
             obj.flags = FLAT_BINDER_FLAG_ACCEPTS_FDS;
-            if (local->isRequestingSid()) {
+            // Disabled for Halium
+            /*if (local->isRequestingSid()) {
                 obj.flags |= FLAT_BINDER_FLAG_TXN_SECURITY_CTX;
-            }
+            }*/
             if (local->isInheritRt()) {
                 obj.flags |= FLAT_BINDER_FLAG_INHERIT_RT;
             }
diff --git a/libs/binder/ProcessState.cpp b/libs/binder/ProcessState.cpp
index fb2781be59..47a7117244 100644
--- a/libs/binder/ProcessState.cpp
+++ b/libs/binder/ProcessState.cpp
@@ -209,7 +209,8 @@ bool ProcessState::becomeContextManager()
     std::unique_lock<std::mutex> _l(mLock);
 
     flat_binder_object obj {
-        .flags = FLAT_BINDER_FLAG_TXN_SECURITY_CTX,
+        // Disabled for Halium
+        /*.flags = FLAT_BINDER_FLAG_TXN_SECURITY_CTX,*/
     };
 
     int result = ioctl(mDriverFD, BINDER_SET_CONTEXT_MGR_EXT, &obj);
-- 
2.55.0

